src/entities/createPool/createPoolV3/weighted/createPoolWeighted.ts

```diff
@@ -49,7 +49,7 @@ export class CreatePoolWeightedV3 implements CreatePoolBase {
         const roleAccounts: PoolRoleAccounts = {
             pauseManager: input.pauseManager,
             swapFeeManager: input.swapFeeManager,
-            poolCreator: input.poolCreator,
+            poolCreator: zeroAddress,
```

Comment:
This one was pretty interesting. The SDK was able to somehow complete the tx to create a weighted pool, but when [simulating with exact same calldata on tenderly](https://dashboard.tenderly.co/balancer/v3/simulator/bac78934-f91a-4925-a118-da342cd4ac6a?trace=0.0), the tx reverted because the monorepo's `WeightedPoolFactory` requires `roleAccounts.poolCreator` to be the zero address.

https://github.com/balancer/balancer-v3-monorepo/blob/98b9efd5d1446ec6e80bb3893374d51601255375/pkg/pool-weighted/contracts/WeightedPoolFactory.sol#L64-L66

Reply:
I wasn't aware of that change! Does indeed seem strange that the SDK tx wasn't reverting.
I think it might be a bit confusing to a user that we allow them to pass a non-zero poolCreator address but it doesn't get used. Maybe we should throw an error if the input isn't zero for this case or change that type?

Reply:
If poolCreator does not exist on WeightedPool (and maybe others), we could/should remove it from `CreatePoolV3BaseInput` and use it only on pool types that actually require it.

Reply:
Agreed good thinking @brunoguerios ! :brain: